be/src/exec/schema_scanner/schema_partitions_meta_scanner.cpp

```diff
@@ -79,12 +79,25 @@ Status SchemaPartitionsMetaScanner::start(RuntimeState* state) {
             auth_info.__set_user_ip(*(_param->user_ip));
         }
     }
-    TGetPartitionsMetaRequest partitions_meta_req;
-    partitions_meta_req.__set_auth_info(auth_info);
 
     // init schema scanner state
     RETURN_IF_ERROR(SchemaScanner::init_schema_scanner_state(state));
-    RETURN_IF_ERROR(SchemaHelper::get_partitions_meta(_ss_state, partitions_meta_req, &_partitions_meta_response));
+    int64_t table_id_offset = 0;
+    while (true) {
+        TGetPartitionsMetaRequest partitions_meta_req;
+        partitions_meta_req.__set_auth_info(auth_info);
+        partitions_meta_req.__set_start_table_id_offset(table_id_offset);
+        TGetPartitionsMetaResponse partitions_meta_response;
+        RETURN_IF_ERROR(SchemaHelper::get_partitions_meta(_ss_state, partitions_meta_req,
```

Comment:
it can be implemented in streaming style, which can reduce the peak memory consumption

Reply:
wait for you to try it on mv